### PR TITLE
[Proposal] Add vendoring for Mapbox.framework to npm package

### PR DESCRIPTION
**Pull request by kitten**
During installation the entire `Mapbox.framework` is downloaded using `curl`. Unfortunately it fails quite easily on unstable network connections, and retrying to download starts over completely.

Instead this entire problem can be side-stepped by vendoring `Mapbox.framework`. It doesn't have to be added to the repo, as before, but we can add it to the npm package.

~Additionally this PR fixes the `podspec` file to point to the framework's headers, which was missing before.~ (Moved to #607)

### CODE REVIEW

**Comment by henrikra** [comment]
I think this helps with https://github.com/mapbox/react-native-mapbox-gl/issues/593

File: .npmignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ android/gradle/
 android/gradlew
 android/gradlew.bat
 android/local.properties
+
+# Vendor iOS Mapbox.framework
+!ios/Mapbox.framework
```
Comment (henrikra) [inline .npmignore:42]: Why `!` is there?

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "url": "https://github.com/mapbox/react-native-mapbox-gl"
   },
   "scripts": {
-    "preinstall": "node ./scripts/download-mapbox-gl-native-ios-if-on-mac.js 3.3.1",
+    "preversion": "./scripts/download-mapbox-gl-native-ios.sh 3.3.1",
```
Comment (henrikra) [inline package.json:25]: What is different with `preinstall` and `preversion`?

File: .npmignore (hunk 1 above)
Comment (kitten) [reply .npmignore:42]: This includes the framework in the tarball that will be sent to the npm registry

File: package.json (hunk 2 above)
Comment (kitten) [reply package.json:25]: This runs the hook before a version is published (npm version) instead of when the user installs the package

Comment (henrikra) [reply package.json:25]: Ah okay! This is good!

Comment (henrikra) [reply package.json:25]: Also if the `download-mapbox-gl-native-ios.sh` fails it won't publish new version?

Comment (kitten) [reply package.json:25]: @henrikra I haven't checked whether it actually aborts 🤔  I'll have to check that

**Comment by danieljvdm** [comment]
What is the status on this PR? I think it's pretty necessary to fix a lot of linking errors.

**Comment by kitten** [comment]
@danieljvdm afaik this repo is currently in a small break of maintenance, but I'll see if I can contact some people in the meantime

**Comment by nitaliano** [comment]
@philpl This is a good idea! We can work this into the repo once we start it's refactor.  For now I think we should hold off on merging this in and just merge in the podspec portion.

**Comment by kitten** [comment]
@nitaliano if you’d like I can move the pod spec fixes into a separate PR?

**Comment by nitaliano** [comment]
yes that would be great if you can

**Comment by kitten** [comment]
@nitaliano :smile: This PR doesn't include the podspec modifications anymore. Those are now in a separate PR here: https://github.com/mapbox/react-native-mapbox-gl/pull/607

**Comment by henrikra** [comment]
@philpl There is still one conflict

**Pull request closed without merging**